plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduPageSink.java

```diff
@@ -189,6 +207,57 @@ else if (type instanceof DecimalType) {
         }
     }
 
+    @Override
+    public void storeMergedRows(Page page)
+    {
+        // The last channel in the page is the rowId block, the next-to-last is the operation block
+        int columnCount = columnTypes.size();
+        checkArgument(page.getChannelCount() == 2 + columnCount, "The page size should be 2 + columnCount (%s), but is %s", columnCount, page.getChannelCount());
+        Block operationBlock = page.getBlock(columnCount);
+        Block rowIds = page.getBlock(columnCount + 1);
+
+        Schema schema = table.getSchema();
+        for (int position = 0; position < page.getPositionCount(); position++) {
+            int operation = operationBlock.getInt(position, 0);
```

Comment:
```java
long operation = INTEGER.getLong(operationBlock, position);

Reply:
Changed here and everywhere else I could find.